src/main/java/de/tum/in/www1/artemis/domain/quiz/DragAndDropQuestion.java

```diff
@@ -176,16 +157,20 @@ public void afterCreate() {
         }
     }
 
-    @PreUpdate
-    public void onUpdate() {
-        // move file and delete old file if necessary
-        backgroundFilePath = fileService.manageFilesForUpdatedFilePath(prevBackgroundFilePath, backgroundFilePath, FilePathService.getDragAndDropBackgroundFilePath(), getId());
-    }
-
+    /**
+     * This method is called when deleting the entity. It makes sure that the corresponding file is deleted as well.
+     */
     @PostRemove
     public void onDelete() {
         // delete old file if necessary
-        fileService.manageFilesForUpdatedFilePath(prevBackgroundFilePath, null, FilePathService.getDragAndDropBackgroundFilePath(), getId());
+        try {
+            if (backgroundFilePath != null) {
+                fileService.deleteFiles(List.of(Path.of(fileService.actualPathForPublicPathOrThrow(backgroundFilePath))));
+            }
+        }
+        catch (FilePathParsingException ignored) {
+            // if the file path is invalid, we don't need to delete it
```

Comment:
Maybe still log this since we might have the file saved somewhere else due to some bug?